### ISSUE TITLE
Spotify as bloatware solution

### ISSUE THREAD

**Issue by sorryusernameisalreadytaken** [sergiocastell/spotify-dogfood#15]
Is there a chance that you push a special verions with, for example a yellow spotify icon and another name (com.spotify2.XYZ)? That would be awesome.

Some people out there dont want to root there phone but got a spotify app as a pre-installed app on there phones. For example many sony phones. When they want to install your modded apk, the got an error, caused by the the problem that they allready own the newest apk.

I think there coud be 2 solutions:
-By changing the app internal package name. (com.spotify2.XYZ)
-By faking a higher version of the app. With the hope that the system overide the old, pre-installed spotify. Perhaps this fails due to right issues. 

Greetings by uniat

**Comment by Seluj78** [comment]
The thing is, by renaming the package, you'd loose a ton of features, because if I understand this right, some apps recognises each other by packages name

**Comment by sorryusernameisalreadytaken** [comment]
Damn it... That means on many devices with pre-installed spotify app as system-app, the can not use this great apk :-(

THX for your reply.

What is about faking a higher version?

**Comment by Seluj78** [comment]
I just want to disclaim that I am not the maintainer of the app, just a random dude who happens to be subscribed to this repository, who uses the app on a daily basis and does some programming to

Anyway, I was saying this based on my understanding, and I don't want to give you too much hope on it. Sergio will answer shortly I think

Have a nice day :smile:

**Comment by Bockwurst-Mit-Puls** [comment]
On many phones you can change the preloaded apps set through some vendor specific trigger. Sony has country/carrier specific preload sets "CSC". Also they are on a different partition than system. You might be able to get rid of it.

But seriously nowadays you should get yourself root access, everything else is kind of irresponsible, considering, that if you dont someone else does/has/will be.


-------- Ursprüngliche Nachricht --------
Von: sorryusernameisalreadytaken <notifications@github.com>
Gesendet: 26. Oktober 2017 09:30:47 MESZ
An: sergiocastell/spotify-dogfood <spotify-dogfood@noreply.github.com>
CC: Subscribed <subscribed@noreply.github.com>
Betreff: Re: [sergiocastell/spotify-dogfood] Spotify as pre-installed app - solution (#15)

Damn it... That means on many devices with pre-installed spotify app as system-app, the can not use this great apk :-(

THX for your reply.

What is about faking a higher version?

**Comment by sergiocastell** [comment]
@sorryusernameisalreadytaken 
> -By faking a higher version of the app. With the hope that the system overide the old, pre-installed spotify. Perhaps this fails due to right issues.

To fake a higher version of the app, I'd still need to trip the application signature, even if I didn't make any code or xml changes. And the reason why you can't install Dogfood on top of a official build is precisely that - there's a signature mismatch between both apk's and Android aborts the installation. 

However, as @Bockwurst-Mit-Puls commented, you can check if Spotify isn't in your actual system partition but sideloaded elsewhere and you can just get rid of it, and then be able to replace it for Spotify Dogfood. 

Of course, not to mention @Seluj78's comment were absolutely right, and making a different, whole new package name for *each release* is time costy for me, and that version would lose features such as Spotify Connect, Android Auto, Wear, etc.  

This issue will be closed if your questions were succesfully answered.

**Comment by fabyGT** [comment]
To install it on a Spotify-pre-installed-device you've only to download Spotify dogfood, download app cloner, open last one, select apk, select Spotify dogfood, follow the instructions and install.
Simple, isn't it?

**Comment by cornaro** [comment]
I had the same problem with my xperia z5 compact but the only solution i found with this problem was to download the app "App Cloner" from the market .Its free. It is the unique application which allows you to make a backup copy from any apk which exist in stock into your sony device and after to trick it reinstall the backup ,but in reality you will install the modded copy you like to insert.Try it and you will be surprised how easy it is.

**Comment by penn5** [comment]
Don't use "App Cloner", it changes package name. Use "Island" by Oasis Feng.

**Comment by tletum13** [comment]
The thing is, by renaming the package, you'd loose a ton of features, because if I understand this right, some apps recognises each other by packages name

**Comment by tletum13** [comment]
changing the app internal package name. (com.spotify2.XYZ)
 faking a higher version of the app. With the hope that the system overide the old, pre-installed spotify. Perhaps this fails due to right issues.

**Comment by Seluj78** [comment]
Thanks @tletum13 for stealing `The thing is, by renaming the package, you'd loose a ton of features, because if I understand this right, some apps recognises each other by packages name` ...

And i don't think `com.spotify2.XYZ` will work either, I'm sure Sergio tried it already

**Comment by StashCat** [comment]
This could be done with the use of Magisk, since the people working on the magisk port of injected youtube playback made it all work. If you already have YouTube installed you either uninstall it or reset to factory app, which means they probably swap the factory apk (or add it) so that the same package name can remain.

**Comment by shinji257** [comment]
Yes but then you have root at this point making the point of this inquiry moot.

**Comment by IamFicticio** [comment]
@penn5 Can't install apk with Island, could you please expand further?

**Comment by penn5** [comment]
Installing in a work profile should work, AFAIK. However, you would have to
move all the appd that you want to talk to spotify dogfood into the work
profile. In the work profile, you can install spotify, as it won't be
pre-installed. Then you can install the spotify that you want, as well as
android auto.

**Comment by IamFicticio** [comment]
Im still unable to install the dogfood version or either I don't understand the app.
When use Island its clones somes app, but nowhere where I can install the dogfood .apk. What I had tried is clone spotify then launch the apk, but as expected "app not installed"

**Comment by penn5** [comment]
I think you would have to clone your file manager, web browser and package installer, and reperform install from scratch in island.

Sent from Mail for Windows 10

**Comment by IamFicticio** [comment]
@penn5 the app only allow me to clone normal apps, nor "functions" like package installer.
Did you use Island to clone your Spotify?
Please someone help.
